### PR TITLE
add controller,mapper,repository imagenews

### PR DESCRIPTION
**Pull request by Cheephot**

### CODE REVIEW

File: src/ImageService.Business/Commands/ImageNews/GetImageNewsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        public OperationResultResponse<ImageNewsResponse> Execute(Guid parentId)
+        {
+            OperationResultResponse<ImageNewsResponse> response = new();
+
+            DbImagesNews dbImageNews = _imageNewsRepository.Get(parentId);
+
+            if (dbImageNews == null)
+            {
+                response.Body = null;
+                response.Errors.Add("Image was not found.");
```
Comment (ndoriomedov) [inline src/ImageService.Business/Commands/ImageNews/GetImageNewsCommand.cs:33]: add status maybe?

**Review by ndoriomedov — APPROVED** [review]

File: src/ImageService.Business/Commands/ImageNews/GetImageNewsCommand.cs (hunk 1 above)
Comment (Cheephot) [reply src/ImageService.Business/Commands/ImageNews/GetImageNewsCommand.cs:33]: done

File: src/ImageService.Mappers/Db/Interfaces/IDbImageNewsMapper.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using LT.DigitalOffice.ImageService.Models.Db;
+using LT.DigitalOffice.Kernel.Attributes;
+using System;
+
+namespace LT.DigitalOffice.ImageService.Mappers.Db.Interfaces
+{
+    [AutoInject]
+    public interface IDbImageNewsMapper
+    {
+        public DbImagesNews Map(Guid parentId, string name, string content, string extension, Guid createdBy);
```
Comment (BelehovEgor) [inline src/ImageService.Mappers/Db/Interfaces/IDbImageNewsMapper.cs:10]: remove public

File: src/ImageService.Data/Interfaces/IImageNewsRepository.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace LT.DigitalOffice.ImageService.Data.Interfaces
+{
+    [AutoInject]
+    public interface IImageNewsRepository
+    {
+        List<Guid> Create(List<DbImagesNews> imagesNews);
+        List<DbImagesNews> Get(List<Guid> imageIds);
+        DbImagesNews Get(Guid imageId);
+        bool Delete(DbImagesNews imagesNews);
```
Comment (VLiamin) [inline src/ImageService.Data/Interfaces/IImageNewsRepository.cs:14]: imageNews. in other cases same name.

File: src/ImageService.Mappers/Responses/ImageDataResponseMapper.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+using LT.DigitalOffice.ImageService.Models.Db;
+using LT.DigitalOffice.ImageService.Models.Dto.Responses;
+
+namespace LT.DigitalOffice.ImageService.Mappers.Models
+{
+    public class ImageDataResponseMapper : IImageDataResponseMapper
+    {
+        public ImageDataResponse Map(DbImagesNews dbImagesNews)
+        {
+            return new ImageDataResponse
```
Comment (VLiamin) [inline src/ImageService.Mappers/Responses/ImageDataResponseMapper.cs:11]: check dbImagesNews != null

File: src/ImageService.Business/Commands/ImageNews/GetImageNewsCommand.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+using LT.DigitalOffice.Kernel.Responses;
+using System;
+
+namespace LT.DigitalOffice.ImageService.Business.Commands.ImageNews
+{
+    public class GetImageNewsCommand : IGetImageNewsCommand
+    {
+        private readonly IImageNewsRepository _imageNewsRepository;
+        private readonly IImageDataResponseMapper _imageNewsResponseMapper;
+        public GetImageNewsCommand(
```
[the remark this answers is not in the document]
Comment (Cheephot) [reply src/ImageService.Business/Commands/ImageNews/GetImageNewsCommand.cs:16]: done

File: src/ImageService.Business/Commands/ImageNews/Interfaces/IGetImageNewsCommand.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+using LT.DigitalOffice.Kernel.Attributes;
+using LT.DigitalOffice.Kernel.Responses;
+using System;
+
+namespace LT.DigitalOffice.ImageService.Business.Commands.ImageNews.Interfaces
+{
+    [AutoInject]
+    public interface IGetImageNewsCommand
+    {
+        OperationResultResponse<ImageDataResponse> Execute(Guid parentId);
```
[the remark this answers is not in the document]
Comment (Cheephot) [reply src/ImageService.Business/Commands/ImageNews/Interfaces/IGetImageNewsCommand.cs:11]: done

File: src/ImageService.Data/Interfaces/IImageNewsRepository.cs (hunk 3 above)
Comment (Cheephot) [reply src/ImageService.Data/Interfaces/IImageNewsRepository.cs]: Let's update the names in another task

File: src/ImageService.Mappers/Responses/ImageDataResponseMapper.cs (hunk 4 above)
Comment (Cheephot) [reply src/ImageService.Mappers/Responses/ImageDataResponseMapper.cs]: Done

File: src/ImageService.Mappers/Responses/ImageDataResponseMapper.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+            {
+                return new ImageDataResponse
+                {
+                    Id = dbImagesNews.Id,
+                    Content = dbImagesNews.Content,
+                    Name = dbImagesNews.Name,
+                    Extension = dbImagesNews.Extension
+                };
+            }
+            else
```
[the remark this answers is not in the document]
Comment (Cheephot) [reply src/ImageService.Mappers/Responses/ImageDataResponseMapper.cs:21]: done

**Review by VLiamin — APPROVED** [review]

**Review by MariyaSin — APPROVED** [review]

File: src/ImageService.Broker/Consumers/ImageNews/DeleteImagesNewsConsumer.cs [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+            object response = OperationResultWrapper.CreateResponse(DeleteImages, context.Message);
+
+            await context.RespondAsync<IOperationResult<bool>>(response);
+        }
+
+        private object DeleteImages(IDeleteImagesNewsRequest request)
+        {
+            _repository.Delete(request.ImageIds);
+
+            return true;
```
Comment (MariyaSin) [inline src/ImageService.Broker/Consumers/ImageNews/DeleteImagesNewsConsumer.cs:31]: repo can return false

Comment (Cheephot) [reply src/ImageService.Broker/Consumers/ImageNews/DeleteImagesNewsConsumer.cs]: done

**Review by borovHokage — APPROVED** [review]

**Pull request merged**